_posts/2019-05-20-server-rendering-responsively.md

```diff
@@ -0,0 +1,201 @@
+---
+layout: epic
+title: "Server-Rendering Responsively"
+date: "2019-05-20"
+author: [steve-hicks]
+categories: [react, html, web]
+---
+
+We use server-side rendering (SSR) to deliver every page you hit on [artsy.net](https://artsy.net). We decided on
+using SSR for many reasons, amongst them performance.
+[We wrote about this all the way back in 2013](https://artsy.github.io/blog/2013/11/30/rendering-on-the-server-and-client-in-node-dot-js/)!
+
+We've also built our site using responsive design, so you get a browsing experience optimized for your device.
+
+Combining SSR and responsive design is a non-trivial problem. There are many concerns to manage, and they are
+sometimes in conflict with each other. We server render for performance reasons, but we also want to be sure our
+app is performant when your browser takes over, all while optimizing for accessibility and SEO.
+
+This article describes the tools we use on [artsy.net](https://artsy.net) to combine SSR and responsive design.
+
+<!-- more -->
+
+## Tool 1: [`styled-system`](https://styled-system.com)
+
+We handle the majority of responsive styling differences with
+[`styled-system`](https://styled-system.com/responsive-styles). This has been a really great addition to our
+toolbox. Here's a component that would render a `div` (`Box`) with a width of 50% at our mobile breakpoint, 75% at
+our tablet breakpoint, and 100% for anything larger:
+
+```xml
+<Box width={["50%", "75%", "100%"]}>
+  ...
+</Box>
+```
+
+Here's a real example
+[from our codebase](https://github.com/artsy/reaction/blob/f3dabb884d616c5c42bbd47b1432dc2a9456b8ca/src/Apps/Search/Components/SearchResultsSkeleton/Header.tsx#L6).
+
+```xml
+<Box height={100} mb={30} mt={120} pr={[0, 20]} pl={[0, 20]}>
+  <Box pl={[20, 0]}>
+    <Box mb={40} background={color("black10")} width={260} height={20} />
+    ...
+  </Box>
+</Box>
+```
+
+There are a lot of abbreviations in there - `mb` sets the bottom margin, `mt` sets the top margin, `pl` sets the
```

Comment:
styled-system v4 allows for actual (long-form) words, so I think it might be better to update the example and let the code speak for itself, instead of having to add explanations here.

https://github.com/styled-system/styled-system/blob/master/CHANGELOG.md#v400-2019-02-23

Reply:
Sounds like an improvement. I'll make this change.